apps/files/src/helpers/collaborators.js

```diff
@@ -0,0 +1,74 @@
+import roles from './collaboratorRoles'
+
+/**
+ * Bitmask of ownCloud permission
+ */
+export const permissionsBitmask = {
+  read: 1,
+  update: 2,
+  create: 4,
+  delete: 8,
+  share: 16,
+  all: 31
+}
+
+/**
+ * Maps role to its bitmask
+ * @param {object} role  Role which is to be mapped to permissions
+ * @param {array} additionalPermissions  Array of selected additional permissions
+ * @returns {number}  Role bitmask
+ */
+export function roleToBitmask (role, additionalPermissions = []) {
+  let bitmask = 0
+
+  for (const permission of role.permissions) {
+    bitmask += permissionsBitmask[permission]
+  }
+
+  if (additionalPermissions) {
+    for (const additionalPermission of additionalPermissions) {
+      bitmask += permissionsBitmask[additionalPermission]
+    }
+  }
+
+  return bitmask
+}
+
+/**
+ * Maps bitmask to role
+ * @param {number} bitmask Permissions which are to be mapped to role
+ * @param {boolean} isFolder Defines if the item is folder
+ * @returns {object} Role mapped to the bitmask
+ */
+export function bitmaskToRole (bitmask, isFolder) {
+  const currentRoles = roles({ isFolder: isFolder })
```

Comment:
add comment "not passing in translation as we don't need it"

Reply:
optional: inject the result of "roles()" in the function header and have the caller of bitmaskToRole call `roles()` with appropriate arguments including translation.

=> add this as a TODO for later